### PR TITLE
New Device, Canopy Motorola 58600 PTP radio

### PR DESCRIPTION
**Pull request by barryodonovan**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/polling/os/cambium.inc.php [hunk 1]
```diff
@@ -20,6 +20,10 @@
     $masterSlaveMode = ucfirst(snmp_get($device, 'masterSlaveMode.0', '-Oqv', "CAMBIUM-PTP250-MIB"));
     $hardware = 'PTP 250 '.$masterSlaveMode;
     include 'includes/polling/wireless/cambium-250.inc.php';
+} elseif (strstr($cambium_type, 'PTP 58600')) {
+    $masterSlaveMode = ucfirst(snmp_get($device, 'masterSlaveMode.0', '-Oqv', "CANOPY-SYS-MIB"));
+    $hardware = 'PTP 58600 '.$masterSlaveMode;
+    include 'includes/polling/wireless/motorola58600.inc.php';
```
Comment (laf) [inline includes/polling/os/cambium.inc.php]: Is this supposed to be there?

File: includes/definitions/cambium.yaml [hunk 2]
```diff
@@ -7,8 +7,9 @@ over:
 group: cambium
 discovery:
     - sysObjectId:
-        - .1.3.6.1.4.1.17713.21
+        - .1.3.6.1.4.1.17713
```
Comment (laf) [inline includes/definitions/cambium.yaml]: So changing this will pick up:

`Motorola PTP 54400 Full, Version 54400-09-02`

Is this a Cambium device?

From our stats, the `'/^Cambium/'` would pick up Cambium PTP 58600 Lite PTP Wireless Ethernet Bridge, Version 58600-10-08

If you can add a test unit that would help.

Comment (barryodonovan) [reply includes/definitions/cambium.yaml]: Sorry, don't follow, where is `Motorola PTP 54400 Full, Version 54400-09-02` picked up?
I couldn't find any Cambium/Motorola PTP except the 250 and the 650 in Librenms.

Cambium is the 'new' name for a division of Motorola Wireless, the bit that Google didn't buy. They rebadged some firmwares, but if a device was EOL, they didn't issue a rebadged firmware, they do still provide hardware support to some extent though, and I guess if they acknowledged a big enough security hole to be worth patching, it would probably come on a rebadged firmware.

Basically however, enterprises.17713 used to be Motorola, now it's Cambium.

Forgot about the unit test...

Comment (barryodonovan) [reply includes/definitions/cambium.yaml]: .1.3.6.1.4.1.17713 is the Cambium backhaul wireless OID, 
.1.3.6.1.4.1.17713.21 is one particular hardware variant.

Comment (barryodonovan) [reply includes/definitions/cambium.yaml]: Ah, I see `Cambium PTP 58600 Lite PTP Wireless Ethernet Bridge, Version 58600-10-08` now.
Odd, there doesn't seem to be an OS to match that test...

Comment (barryodonovan) [reply includes/definitions/cambium.yaml]: So... this is turning stream of conciousness on my part...
.1.3.6.1.4.1.17713.21 is an ePMP device. The cambium.snmprec sysDescr uses the string, `WaveDirect` which is the name of the company who did the original discovery.
They added discovery for the various pieces of hardware they were using, in a rather haphazard manner. the Motorola/Cambium MIBS seem to be well written, they are broken down by hardware variant, then they follow a consistent structure wherever possible so I think things can be much better.

I'd like to have time to rewrite the entire Cambium / Canopy / Motorola discovery and polling to reflect the new wireless paradigm and achieve some consistency across the product line but I have two problems:
1. I am quite unclear about the structure and concepts of the new wireless paradigm;
2. I have about 10 other responsibilities beyond getting LibreNMS integrated in house...

Is there a reasonably concise guide to, or description of the wireless paradigm?
Can I simply tidy this up for now to get things working and open a new issue / PR to re-factor Cambium family discovery later?

Comment (laf) [reply includes/definitions/cambium.yaml]: So `Motorola PTP 54400 Full, Version 54400-09-02` is from the stats we get from users.

I'll leave it up to you (or someone who knows this vendor better) to decide the best way forward. I was merely pointing out that the sysObjectID you've changed to will also pick up that motorola device so if that's not expected then you can't use that sysObjectID on it's own.

The wireless support has been revamped, some old devices which were shoe horned in where they could be haven't been updated yet. If you want to add more / extend the wireless stats for an OS then follow the new docs on that and ignore what was there before.

Comment (murrant) [reply includes/definitions/cambium.yaml]: Yes, all of these devices are no longer Motorola they are Cambium.

However, it makes me sad that when I finally get around to converting these to the new code, this is just one more thing to convert.

**Comment by barryodonovan** [comment]
I know, sorry :-(

I've been trying to get my head around the new code to no avail...

There's talk further up the thread of refactoring the whole cambium shebang using the new paradigm. I thought I might have gotten to do some of it this week but I was only in the office for one day and this was way down the list of priorities!

@laf mentioned some docs for the new approach to wireless but I couldn't find them.
If you could point me at them I'd appreciate it
________________________________

**Comment by laf** [comment]
> @laf mentioned some docs for the new approach to wireless but I couldn't find them.
> If you could point me at them I'd appreciate it

https://docs.librenms.org/#Developing/os/Wireless-Sensors/

**Comment by barryodonovan** [comment]
Thanks.

________________________________

**Comment by pheinrichs** [comment]
@barryodonovan  I've come along a pretty far way in PR #7703 Might be easier to just implement this in my PR, if you give me the full sysObjectId i can add discovery and the graphs for this that you've implemented.

**Comment by barryodonovan** [comment]
Hi Paul,


Looks good, haven't had any time to look at it since so really happy you can do so.

I'll close this PR and help as I can on your one.

________________________________

File: includes/polling/os/cambium.inc.php (hunk 1 above)
Comment (YukonRob) [reply includes/polling/os/cambium.inc.php]: Hopefully not hijacking but Cambium PTP 58500 are showing as generic

Comment (pheinrichs) [reply includes/polling/os/cambium.inc.php]: @YukonRob this is expected. We'll need to add support for the 58500, free free to open a new issue and add the requested files and we can do that for you.
